Source/ARTRealtimeChannel.m

```diff
@@ -305,6 +305,17 @@ - (BOOL)canBeReattached {
     }
 }
 
+- (BOOL)shouldAttach {
+    switch (self.state_nosync) {
+        case ARTRealtimeChannelInitialized:
+        case ARTRealtimeChannelDetaching:
```

Comment:
is not best to wait for it finish detaching before attaching?

Reply:
Sure, it's implemented in `internalAttach:`.